readme.md

```diff
@@ -691,7 +697,85 @@ coordsToPair(pairToCoords([1, 2])) // [1, 2]
 pairToCoords(coordsToPair({x: 1, y: 2})) // {x: 1, y: 2}
 ```
 
+### Homomorphism
 
+A homomorphism is just a structure preserving map. In fact, a functor is just a homomorphism between categories as it preserves the original category's structure under the mapping.
+
+```js
+A.of(f).ap(A.of(x)) == A.of(f(x))
+
+Either.of(_.toUpper).ap(Either.of("oreos")) == Either.of(_.toUpper("oreos"))
+```
+
+### Catamorphism
+
+A `reduceRight` function that applies a function against an accumulator and each value of the array (from right-to-left) to reduce it to a single value.
```

Comment:
@jethrolarson exactly; catamorphism can be seen on multiple monadic types with  or without Foldable spec

```
Identity.of(1).get(); //=> catamorphism
Identity.of(1).fold((acc, val) => acc + val, 1); //=> catamorphism
Maybe.fromNull(value).cata(identity, identity); //=> catamorphism